src/Shared/DomainEventWrapper.cs

```diff
@@ -69,7 +69,7 @@ public object Unwrap(Type returnType)
     }
 
     public static DomainEventWrapper Wrap<T>(T domainEvent)
-        where T : IDomainEvent, new()
```

Comment:
Maybe add some test that would fail if someone adds this again in the future?

Reply:
yup, as discussed, changed a DomainEvent signature in a test to initialize a domain event as a record